lib/rules/element-attributes-order.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const Rule = require('./base');
+
+module.exports = class ElementAttributesOrder extends Rule {
+  visitor() {
+    const log = this.log.bind(this);
+    const sourceForNode = this.sourceForNode.bind(this);
+
+    function extractElementTokens(node, nodeText) {
+      const keys = [];
+      node.attributes.forEach(attr => {
+        const source = sourceForNode(attr);
+        const type = attr.name.startsWith('@')
+          ? 'arguments'
+          : attr.name.startsWith('...')
+          ? 'splattributes'
+          : 'attributes';
```

Comment:
A nice-to-have improvement would be to extract the attribute types into an enum:

```js
const AttributeType = {
    ARGUMENTS: 'arguments',
    ATTRIBUTES: 'attributes',
    MODIFIERS: 'modifiers',
    ...
}

// example usage
if (type === AttributeType.ARGUMENTS) { ... }
```

Reply:
good catch! added